Userland/Libraries/LibWeb/CSS/Parser/Parser.cpp

```diff
@@ -1674,6 +1678,116 @@ JS::GCPtr<CSSSupportsRule> Parser::convert_to_supports_rule(Rule& rule)
+        return {};
+    }
+
+    auto const& name = name_token.ident();
+
+    auto block_tokens = rule.block()->values();
+    auto block_stream = TokenStream { block_tokens };
+
+    auto list_of_declarations = parse_a_list_of_declarations(block_stream);
+
+    auto syntax_maybe = Optional<FlyString> {};
+    auto inherit_maybe = Optional<bool> {};
+    auto initial_value_maybe = Optional<FlyString> {};
+
+    for (auto const& declOrRule : list_of_declarations) {
+        if (!declOrRule.is_declaration()) {
+            dbgln_if(CSS_PARSER_DEBUG, "CSSParser: Only declarations allowed in @property block.; discarding.");
+            return {};
+        }
+        auto const& decl = declOrRule.declaration();
+
+        if (decl.name() == "syntax") {
+            if (syntax_maybe.has_value()) {
+                dbgln_if(CSS_PARSER_DEBUG, "CSSParser: Found multiple declarations of \"syntax\" in @property block; discarding.");
+                return {};
+            }
+            if (decl.values().size() == 1 && decl.values()[0].is(Token::Type::String)) {
+                syntax_maybe = decl.values()[0].token().string();
+            } else {
+                dbgln_if(CSS_PARSER_DEBUG, "CSSParser: Unexpected value for @property \"syntax\": {}; discarding.", decl.to_string());
+            }
+        }
+        if (decl.name() == "inherits") {
+            if (inherit_maybe.has_value()) {
+                dbgln_if(CSS_PARSER_DEBUG, "CSSParser: Found multiple declarations of \"inherits\" in @property block; discarding.");
+                return {};
+            }
+            // Feels hacky using Ident here, but this is what the tokenizer gives us.
+            if (decl.values().size() == 1 && decl.values()[0].is(Token::Type::Ident)) {
+                auto const& ident = decl.values()[0].token().ident();
+                if (ident.is_one_of("true", "false")) {
+                    inherit_maybe = (ident == "true");
+                } else {
+                    dbgln_if(CSS_PARSER_DEBUG, "CSSParser: Expected true/false for @property \"inherits\" value, got: {}; discarding.", ident.to_string());
+                }
+
+            } else {
+                dbgln_if(CSS_PARSER_DEBUG, "CSSParser: Unexpected value for @property \"inherits\": {}; discarding.", decl.to_string());
+            }
+        }
+        if (decl.name() == "initial-value") {
+            if (initial_value_maybe.has_value()) {
+                dbgln_if(CSS_PARSER_DEBUG, "CSSParser: Found multiple declarations of \"initalValue\" in @property block; discarding.");
+                return {};
+            }
+            // FIXME: Ensure that the initial value matches the syntax, and parse the correct CSSValue out
+            StringBuilder initial_value_sb;
+            for (auto const& component : decl.values()) {
+                initial_value_sb.append(component.to_string());
```

Comment:
Not a huge fan of this, but I couldn't find a serializer for the rhs of a declaration.

Reply:
Yeah this is basically the only way to do this right now. It's not ideal. But as you noted, storing a string here is temporary anyway.